### PR TITLE
(MAINT) Document how to do ssh agent forwarding.

### PR DESCRIPTION
**Pull request by smortex**
I literally spent hours understanding that the `@option` / `option` hash in the code was coming from the `CONFIG` section of the hosts file…  Such a howto would have saved my day :-)

### CODE REVIEW

File: docs/how_to/ssh_agent_forwarding.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+file:
+
+~~~yaml
+HOSTS:
+  ...
+CONFIG:
+  forward_ssh_agent: true
+~~~
+
+Beaker will then be able to use the user's agent on the host machine.  There is
```
Comment (kevpl) [inline docs/how_to/ssh_agent_forwarding.md]: Please clarify 'host machine' in this line. I'm assuming you mean a "System Under Test (SUT)" (a host in the testing environment), but we use that term or the "beaker coordinator" for the host that beaker is running on.

**Comment by kevpl** [comment]
Thanks for the doc @smortex! I left one quick comment, but other than that, things look good, and I look forward to getting this merged!

**Comment by smortex** [comment]
I tried to improve the wording @kevpl.  Please tell me if other improvements are needed.

Thanks!

**Comment by kevpl** [comment]
Looks great! Thanks so much for the doc @smortex!

**Pull request merged**